src/Microsoft.Identity.Web.TokenAcquisition/TokenAcquisition.cs

```diff
@@ -358,10 +360,40 @@ private void LogAuthResult(AuthenticationResult? authenticationResult)
                 throw new ArgumentException(IDWebErrorMessage.ClientCredentialTenantShouldBeTenanted, nameof(tenant));
             }
 
+            // If using managed identity 
+            if (tokenAcquisitionOptions != null && tokenAcquisitionOptions.ManagedIdentity != null)
+            {
+                IManagedIdentityApplication miApplication;
+
+                if (tokenAcquisitionOptions.ManagedIdentity.ManagedIdentityType == ManagedIdentityType.UserAssigned &&
+                    !tokenAcquisitionOptions.ManagedIdentity.ClientId.IsNullOrEmpty()
```

Comment:
Doesn't it provoke a null reference exception if ClientID is null? How does it work? is it an extension method?

Reply:
The IsNullOrEmpty method is static and built into the String class, among others. It is designed to test for null values so it has no problem being called from a null value.